Core/src/main/java/com/theprogrammingturkey/comz/game/Game.java

```diff
@@ -268,7 +260,26 @@ public Game(String name)
 	 */
 	public PlayerWeaponManager getPlayersWeapons(Player player)
 	{
-		return playersGuns.computeIfAbsent(player, PlayerWeaponManager::new);
+		if (getPlayers().contains(player))
+		{
+			return gamePlayers.get(player).getWeaponManager();
+		}
+		return null;
+	}
+
+	public List<Player> getPlayers() {
```

Comment:
I really like the idea of condensing a lot of the player logic into the GamePlayer class, but this getPlayers method and gamePlayers map are being used interchangeably in the code when they are not the same. Should probably rename this method to be more accurate and make sure the code is accessing the correct data